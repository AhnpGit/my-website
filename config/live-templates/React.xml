<templateSet group="React">
  <template name="rx_module" value="const _ACTION_PREFIX = '$projectName$/$moduleName$'&#10;&#10;const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`&#10;&#10;// TODO: Connect the reducer.&#10;export default function reducer(state = {$defaultState$}, action) {&#10;    switch (action.type) {&#10;        case $ACTION_NAME$:&#10;            return {$END$}&#10;&#10;        default:&#10;            return state&#10;    }&#10;}&#10;&#10;export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: {$payload$},&#10;    meta: {},&#10;    error: undefined,&#10;})&#10;" description="Creates a DUCKS-module" toReformat="true" toShortenFQNames="true">
    <variable name="projectName" expression="" defaultValue="&quot;&lt;project name&gt;&quot;" alwaysStopAt="true" />
    <variable name="moduleName" expression="fileNameWithoutExtension()" defaultValue="" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <variable name="defaultState" expression="" defaultValue="" alwaysStopAt="true" />
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="payload" expression="" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_component" value="import React from 'react'&#10;&#10;const $ComponentName$ = ({$props$}) =&gt; (&#10;    &lt;$END$/&gt;&#10;)&#10;&#10;export default $ComponentName$" description="Creates a functional component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="props" expression="" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_component_class" value="import React from 'react'&#10;&#10;class $ComponentName$ extends React.Component {&#10;    render() {&#10;        return (&#10;            &lt;$END$/&gt;&#10;        )&#10;    }&#10;}&#10;&#10;export default $ComponentName$" description="Creates a class component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_proptypes" value="import PropTypes from 'prop-types'&#10;$ComponentName$.propTypes = {&#10;    $prop$: PropTypes.$END$,&#10;}" description="Adds proptypes to a component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="prop" expression="" defaultValue="&quot;prop&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_action_async" value="const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`&#10;const $ACTION_NAME$_PENDING = `${$ACTION_NAME$}_PENDING`&#10;const $ACTION_NAME$_FULFILLED = `${$ACTION_NAME$}_FULFILLED`&#10;const $ACTION_NAME$_REJECTED = `${$ACTION_NAME$}_REJECTED`" description="Creates async action types" toReformat="true" toShortenFQNames="true">
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_action" value="const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`" description="Creates an action type" toReformat="true" toShortenFQNames="true">
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_creator" value="export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: {$END$},&#10;    meta: {},&#10;    error: undefined,&#10;})" description="Creates an action creator" toReformat="true" toShortenFQNames="true">
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_creator_async" value="export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: getPayload(),&#10;    meta: {},&#10;    error: undefined,&#10;    &#10;    async function getPayload() {&#10;        return $END$&#10;    }&#10;})" description="Creates an async action creator" toReformat="true" toShortenFQNames="true">
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_container" value="import { connect } from 'react-redux'&#10;import $ComponentName$ from '../components/$ComponentName$'&#10;&#10;const mapStateToProps = (state) =&gt; ({&#10;    $END$&#10;})&#10;&#10;const mapDispatchToProps = (dispatch) =&gt; ({&#10;})&#10;&#10;export default connect(&#10;    mapStateToProps,&#10;    mapDispatchToProps,&#10;)($ComponentName$)" description="Creates a container class" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_styled" value="import styled from 'styled-components'&#10;&#10;const $ComponentName$ = styled.$ElementType$`&#10;  $END$&#10;`&#10;&#10;export default $ComponentName$" description="Creates a styled component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="ElementType" expression="" defaultValue="&quot;div&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
</templateSet>
