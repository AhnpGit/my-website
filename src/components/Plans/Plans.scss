@import '../../temp/sass/abstracts/variables';
@import '../../temp/sass/abstracts/mixins';
@import '../../temp/sass/abstracts/functions';

.wrapper__plans {
  background-color: map_get($colors, aquaHaze);

  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  @include mq(nth($medium-up, 1)) {
    padding: 0 map_get($base, unit);
  }

  .plans__intro {
    @extend .wrapper__plans;
    padding: 0;

    h2 {
      text-align: center;
      word-spacing: 2px;
      letter-spacing: 1px;

      &:after {
        display: block;
        height: 2px;
        background-color: map_get($colors, tahitiGold);
        content: " ";
        width: 100px;
        margin: 0 auto;
        margin-top: map_get($base, 'unit');
      }
    }
  }

  @lost clearing both;
  @lost rounder 99.9;
  .plans__body {
    lost-center: 100% 0 flex;
    justify-content: space-between;

    .plans__box {
      lost-row: 1/3 map_get($base, 'unit');
      flex: 1 1 auto;
      display: flex;
      flex-direction: column;
      justify-content: space-around;
      background-color: #fff;
      border-radius: 5px;
      box-shadow: 0 2px 2px #efefef;
    }

    @include mq(nth($medium-up, 1)) {
      .plans__box {
        lost-column: 1/3 0 map_get($base, 'unit');
        height: 351px;
      }
    }
  }
}

.box__header {
  flex: 1 0 auto;

  h3 {
    text-transform: uppercase;
  }

  .price {
    font-size: 120%;
    color: map_get($colors, tahitiGold);
  }

  border-bottom: 1px solid #e8e8e8;
  padding: map_get($base, spacer) 0 0 map_get($base, spacer);
}

.box__body {
  flex: 2 0 auto;
  display: flex;
  flex-direction: column;
  justify-content: space-around;

  padding: map_get($base, unit);
  border-bottom: 1px solid #e8e8e8;
  @include mq(nth($small-only, 1), nth($small-only, 2)) {
    font-size: 85%;
  }

  svg {
    fill: map_get($colors, malachite);
    margin-right: map_get($base, spacer);

    @include stepScale('width', map-deep-get($base, 'icon', 'small'), map-deep-get($base, 'icon', 'small'));
    @include stepScale('height', map-deep-get($base, 'icon', 'small'), map-deep-get($base, 'icon', 'small'));
  }

  .box__body__item {
    display: flex;
    align-items: center ;
    @include mq(nth($medium-up, 1)) {
      font-size: 90%;
    }
  }
}

.box__footer {
  flex: 1 0 auto;
  padding: map_get($base, spacer) 0;

  display: flex;
  justify-content: center;
  align-items: center;
}
