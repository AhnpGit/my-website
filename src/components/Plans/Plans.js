import React from 'react'

import Row from '../../elements/Row/Row'
import Wrapper from '../../elements/Wrapper/Wrapper'
import Button from '../../elements/Button/Button'
import Icon from '../../elements/Icon/Icon'

import './Plans.scss'

/**
 * Plan component displays 3 plans
 */
const Plans = ({ planItems }) => (
  <Row className="plans">
    <Wrapper name="plans">
      <div className="plans__intro">
        <h2>Start eating healthy today</h2>
      </div>
      <div className="plans__body">
        {planItems.map(plan =>
          <div className="plans__box" key={plan.id}>
            <div className="box__header">
              <h3>{plan.title}</h3>
              <p className="price">{plan.price}</p>
              <p>{plan.priceDetail}</p>
            </div>
            <div className="box__body">
              {plan.features.map(detail =>
                <div className="box__body__item" key={detail.id}><Icon name="check" />{detail.description}</div>
              )}
            </div>
            <div className="box__footer">
              <Button text="Sign up now" onClick={() => {}} />
            </div>
          </div>
        )}
      </div>
    </Wrapper>
  </Row>
)

export default Plans
