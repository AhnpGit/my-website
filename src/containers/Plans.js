// @flow

import { connect } from 'react-redux'
import Plans from '../components/Plans/Plans'
import type { PlansType } from '../utils/types'

type PlansState = {
  plans: PlansType
}

export const mapStateToProps = (state : PlansState) => ({
  planItems: state.plans.planItems
})

export default connect(
  mapStateToProps
)(Plans)
