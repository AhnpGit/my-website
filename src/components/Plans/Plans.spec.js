import React from 'react'
import { shallow } from 'enzyme'
import Plans from './Plans'

describe('Plans component', () => {
  it('renders without any issues', () => {
    const planItems = [
      {
        id: 1,
        title: "Premium",
        price: "$399/month",
        priceDetail: "That's only 13.30$ per meal",
        features: [
          {
            id: 1,
            description: "1 meal every day"
          },
          {
            id: 2,
            description: "Order 24/7"
          },
          {
            id: 3,
            description: "Access to newest creations"
          },
          {
            id: 4,
            description: "Free delivery"
          }
        ]
      }
    ]

    const result = shallow(
      <Plans planItems={planItems} />
    )

    expect(result).toMatchSnapshot()
  })
})
