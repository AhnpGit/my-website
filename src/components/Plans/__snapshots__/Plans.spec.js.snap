// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plans component renders without any issues 1`] = `
<Row
  className="plans"
>
  <Wrapper
    name="plans"
  >
    <div
      className="plans__intro"
    >
      <h2>
        Start eating healthy today
      </h2>
    </div>
    <div
      className="plans__body"
    >
      <div
        className="plans__box"
      >
        <div
          className="box__header"
        >
          <h3>
            Premium
          </h3>
          <p
            className="price"
          >
            $399/month
          </p>
          <p>
            That's only 13.30$ per meal
          </p>
        </div>
        <div
          className="box__body"
        >
          <div
            className="box__body__item"
          >
            <Icon
              name="check"
            />
            1 meal every day
          </div>
          <div
            className="box__body__item"
          >
            <Icon
              name="check"
            />
            Order 24/7
          </div>
          <div
            className="box__body__item"
          >
            <Icon
              name="check"
            />
            Access to newest creations
          </div>
          <div
            className="box__body__item"
          >
            <Icon
              name="check"
            />
            Free delivery
          </div>
        </div>
        <div
          className="box__footer"
        >
          <Button
            onClick={[Function]}
            text="Sign up now"
          />
        </div>
      </div>
    </div>
  </Wrapper>
</Row>
`;
